src/llmcompressor/transformers/compression/helpers.py

```diff
@@ -240,3 +248,104 @@ def calculate_offload_device_map(
         del dummy_model
 
     return device_map
+
+
+def infer_sparse_targets_and_ignores(
+    model: torch.nn.Module,
+    sparsity_structure: str,
+    sparsity_threshold: float,
+) -> Tuple[List[str], List[str]]:
+    """
+    Infers the target and ignore layers in the given model
+    to be used for sparsity compression
+
+    :param model: model to check
+    :param sparsity_structure: sparsity structure to check against
+    :param sparsity_threshold: threshold for sparsity
+    :return: tuple of target and ignore layers
+    """
+
+    exhaustive_targets, exhaustive_ignore = _get_sparse_targets_ignore_dicts(
+        module=model,
+        sparsity_structure=sparsity_structure,
+        sparsity_threshold=sparsity_threshold,
+    )
+
+    return _reduce_targets_and_ignores_into_lists(
+        exhaustive_targets=exhaustive_targets,
+        exhaustive_ignore=exhaustive_ignore,
+    )
+
+
+def is_sparse_compression_target(
+    module: torch.nn.Module, sparsity_threshold: float, sparsity_structure: str
+) -> bool:
+    """
+    :param module: module to check
+    :param sparsity_threshold: threshold for sparsity
+    :param sparsity_structure: sparsity structure to check against
+    :return: whether or not the module is a target for sparsity compression,
+        i.e True if it is sparse and follows the sparsity structure, else False
+    """
+    return (
+        hasattr(module, "weight")
+        and tensor_sparsity(module.weight) >= sparsity_threshold
+        and tensor_follows_mask_structure(tensor=module.weight, mask=sparsity_structure)
+    )
+
+
+def _get_sparse_targets_ignore_dicts(
+    module: torch.nn.Module, sparsity_structure: str, sparsity_threshold: float
+) -> Tuple[Dict[str, List[str]], Dict[str, List[str]]]:
+    """
+    Get sparse targets and ignore dictionaries
+
+    :param module: module to check
+    :param sparsity_structure: sparsity structure to check against
+    :param sparsity_threshold: threshold for sparsity
+    :return: tuple of exhaustive targets and ignore dictionaries
+    """
+    exhaustive_targets = defaultdict(list)
+    exhaustive_ignore = defaultdict(list)
+
+    for name, submodule in iter_named_leaf_modules(module):
+        layer_type = module_type(submodule)
+        is_target = is_sparse_compression_target(
+            module=submodule,
+            sparsity_threshold=sparsity_threshold,
+            sparsity_structure=sparsity_structure,
+        )
+        target_dict = exhaustive_targets if is_target else exhaustive_ignore
+        target_dict[layer_type].append(name)
+    return exhaustive_targets, exhaustive_ignore
+
+
+def _reduce_targets_and_ignores_into_lists(
```

Comment:
From the PR description:
```
Note: We also perform a reduction of the targets, and ignore list to make them cleaner and easily readable inside config.json
```

We don't NEED NEED this function, it just makes the compression_config more legible for end users, otherwise the targets and ignores would contain all param names for example without this the compression config would look like:

```json
  "compression_config": {
    "sparsity_config": {
      "format": "sparse-bitmask",
      "global_sparsity": 0.12233772669884009,
      "ignore": [
        "model.embed_tokens.weight",
        "model.layers.0.input_layernorm.weight",
        "model.layers.0.post_attention_layernorm.weight",
        "model.layers.1.input_layernorm.weight",
        "model.layers.1.post_attention_layernorm.weight",
        "model.layers.2.input_layernorm.weight",
        "model.layers.2.post_attention_layernorm.weight",
        "model.layers.3.input_layernorm.weight",
        "model.layers.3.post_attention_layernorm.weight",
        "model.layers.4.input_layernorm.weight",
        "model.layers.4.post_attention_layernorm.weight",
        "model.layers.5.input_layernorm.weight",
        "model.layers.5.post_attention_layernorm.weight",
        "model.norm.weight",
        "lm_head.weight"
      ],
      "registry_requires_subclass": false,
      "sparsity_structure": "2:4",
      "targets": [
        "model.layers.0.self_attn.q_proj.weight",
        "model.layers.0.self_attn.k_proj.weight",
        "model.layers.0.self_attn.v_proj.weight",
        "model.layers.0.self_attn.o_proj.weight",
        "model.layers.0.mlp.gate_proj.weight",
        "model.layers.0.mlp.up_proj.weight",
        "model.layers.0.mlp.down_proj.weight",
        "model.layers.1.self_attn.q_proj.weight",
        "model.layers.1.self_attn.k_proj.weight",
        "model.layers.1.self_attn.v_proj.weight",
        "model.layers.1.self_attn.o_proj.weight",
        "model.layers.1.mlp.gate_proj.weight",
        "model.layers.1.mlp.up_proj.weight",
        "model.layers.1.mlp.down_proj.weight",
        "model.layers.2.self_attn.q_proj.weight",
        "model.layers.2.self_attn.k_proj.weight",
        "model.layers.2.self_attn.v_proj.weight",
        "model.layers.2.self_attn.o_proj.weight",
        "model.layers.2.mlp.gate_proj.weight",
        "model.layers.2.mlp.up_proj.weight",
        "model.layers.2.mlp.down_proj.weight",
        "model.layers.3.self_attn.q_proj.weight",
        "model.layers.3.self_attn.k_proj.weight",
        "model.layers.3.self_attn.v_proj.weight",
        "model.layers.3.self_attn.o_proj.weight",
        "model.layers.3.mlp.gate_proj.weight",
        "model.layers.3.mlp.up_proj.weight",
        "model.layers.3.mlp.down_proj.weight",
        "model.layers.4.self_attn.q_proj.weight",
        "model.layers.4.self_attn.k_proj.weight",
        "model.layers.4.self_attn.v_proj.weight",
        "model.layers.4.self_attn.o_proj.weight",
        "model.layers.4.mlp.gate_proj.weight",
        "model.layers.4.mlp.up_proj.weight",
        "model.layers.4.mlp.down_proj.weight",
        "model.layers.5.self_attn.q_proj.weight",
        "model.layers.5.self_attn.k_proj.weight",
        "model.layers.5.self_attn.v_proj.weight",
        "model.layers.5.self_attn.o_proj.weight",
        "model.layers.5.mlp.gate_proj.weight",
        "model.layers.5.mlp.up_proj.weight",
        "model.layers.5.mlp.down_proj.weight"
      ]
    }
```

Reply:
This is really useful actually... How do we tell if a given string is a layer name or a module/class though?